demos/sample_azure_iot_pnp/sample_azure_iot_pnp.h

```diff
@@ -0,0 +1,107 @@
+ * 
+ * @return uint32_t Zero if successful, non-zero if any failure occurs.
+ */
+uint32_t ulCreateTelemetry( uint8_t * pucTelemetryData,
+                            uint32_t ulTelemetryDataSize,
+                            uint32_t * pulTelemetryDataLength );
+
+/**
+ * @brief Provides the payload to be sent as reported properties update to the Azure IoT Hub.
+ *
+ * @remark This function must be implemented by the specific sample.
+ *         `ulCreateReportedPropertiesUpdate` is called periodically by the sample 
+ *         core task (the task created by `vStartDemoTask`).
+ *         If the sample does not have any properties to update, just return zero to inform no 
+ *         update should be sent.
+ * 
+ * @param[out] pucPropertiesData    Pointer to uint8_t* that will contain the reported properties payload.
+ * @param[in]  ulPropertiesDataSize Size of `pucPropertiesData`
+ * 
+ * @return uint32_t The number of bytes written in `pucPropertiesData`.
+ */
+uint32_t ulCreateReportedPropertiesUpdate( uint8_t * pucPropertiesData,
+                                           uint32_t ulPropertiesDataSize );
+
+/**
+ * @brief Handles a Command received from the Azure IoT Hub.
+ * 
+ * @remark This function must be implemented by the specific sample.
+ *
+ * @param[in]  pxMessage                          Pointer to a structure that holds details of the Command.
+ * @param[out] pulResponseStatus                  Status code to be sent as response for Command request.
+ * @param[out] pucCommandResponsePayloadBuffer    Buffer in which to write a payload for the Command response.
+ * @param[in]  ulCommandResponsePayloadBufferSize Total size of `ucCommandResponsePayloadBuffer`.
+ * 
+ * @return uint32_t Number of bytes written to `ucCommandResponsePayloadBuffer`.
+ */
+uint32_t ulHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                          uint32_t* pulResponseStatus,
+                          uint8_t* pucCommandResponsePayloadBuffer,
+                          uint32_t ulCommandResponsePayloadBufferSize );
+
+/**
+ * @brief Provides the payload to be sent as telemetry to the Azure IoT Hub.
+ *
+ * @remark This function must be implemented by the specific sample.
+ *
+ * @param[in] pxMessage Pointer to a structure that holds the Properties received.
+ * @param[in] pvContext Context defined by the sample.
+ */
+void vHandleProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                        void * pvContext );
+
+/**
+ * @brief Sends a reported properties update to Azure IoT Hub.
+ * 
+ * @remark This function can be used by a specific sample to update the device's reported properties.
+ *
+ * @param[in] pucProperties      Reported properties content to be updated.
+ * @param[in] ulPropertiesLength Length of `ucProperties`.
+ * 
+ * @return AzureIoTResult_t Result of updating properties.
+ */
+AzureIoTResult_t xUpdateProperties( uint8_t* pucProperties, 

```

Comment:
>xUpdateProperties [](http://example.com/codeflow?start=17&length=17)

Should we add send in name of this API... hard to know that this API sends property.

Reply:
Sure. Renamed to xSendPropertiesUpdate.